### ISSUE TITLE
[BUG] Export with `export_media=False` overwrites original data directory

### ISSUE THREAD

**Issue by dahliarobotics** [voxel51/fiftyone#2084]
### System information

-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 20.04
-   **FiftyOne installed from (pip or source)**: pip
-   **FiftyOne version (run `fiftyone --version`)**: FiftyOne v0.16.5,
-   **Python version**: Python 3.8.10

### Commands to reproduce

```
import fiftyone.types as fot

path = "/some/path"
# dataset
# - is loaded with ImageSegmentationDirectory
# - the ground truth segmentation mask is saved under the label_field "segmentation"
# - has proper mask with array of type uint8

label_type = "segmentation"
dataset_type = fot.ImageSegmentationDirectory
dataset.export(
    export_dir=path,
    dataset_type=dataset_type,
    label_field=label_type,
    data_path="data",
    labels_path=label_type,
    export_media=False,
    overwrite=False,
)
```

### Describe the problem

When exporting a dataset of type `ImageSegmentationDirectory` with the parameter `export_media=False`
- the output path stays empty
- the directory with the original data from the dataset gets corrupted: the images in the folder with the original raw images (e.g. `mydataset/data`) get overwritten with the segmentation masks.
In effect, the folder with the original data is destroyed and the original data lost.

### What areas of FiftyOne does this bug affect?

-   [ ] `App`: FiftyOne application issue
-   [x] `Core`: Core `fiftyone` Python library issue
-   [ ] `Server`: Fiftyone server issue

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another
member of your organization be willing to contribute a fix for this bug to the
FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently.
-   [ ] Yes. I would be willing to contribute a fix for this bug with guidance
        from the FiftyOne community.
-   [x] No. I cannot contribute a bug fix at this time.

**Comment by brimoor** [comment]
I'm not quite following what you're saying the issue is, but the `ImageSegmentationDirectory` exporter is  working as expected.

In the example below, `dataset2` is a segmentation dataset with its contents written to `/tmp/segmentations` on disk, but there's no issue after exporting (only) its masks using `export(..., export_media=False)`

```py
import fiftyone as fo
import fiftyone.zoo as foz

TEST_DIR = "/tmp/segmentations"

dataset = foz.load_zoo_dataset(
    "coco-2017",
    split="validation",
    label_types="segmentations",
    max_samples=51,
)

dataset.export(
    export_dir=TEST_DIR,
    dataset_type=fo.types.ImageSegmentationDirectory,
    overwrite=True,
)

"""
$ ls /tmp/segmentations
data    labels

$ ls /tmp/segmentations/data | head -3
000000000139.jpg
000000000285.jpg
000000000632.jpg

$ ls /tmp/segmentations/labels | head -3
000000000139.png
000000000285.png
000000000632.png
"""

dataset2 = fo.Dataset.from_dir(
    dataset_dir=TEST_DIR,
    dataset_type=fo.types.ImageSegmentationDirectory,
)

# Now export only the masks to a new `segmentations/` subdirectory of `TEST_DIR`

dataset2.export(
    export_dir=TEST_DIR,
    dataset_type=fo.types.ImageSegmentationDirectory,
    labels_path="segmentations",
    export_media=False,
)

"""
$ ls /tmp/segmentations
data        labels      segmentations

$ ls /tmp/segmentations/data | head -3
000000000139.jpg
000000000285.jpg
000000000632.jpg

$ ls /tmp/segmentations/labels | head -3
000000000139.png
000000000285.png
000000000632.png

$ ls /tmp/segmentations/segmentations | head -3
000000000139.png
000000000285.png
000000000632.png
"""

# The dataset still loads fine in the App
session = fo.launch_app(dataset2)
```

**Comment by dahliarobotics** [comment]
Sorry, I think I must be more specific:

### Case 1: Export to a new `segmentations/` subdirectory of `TEST_DIR`
- everything works fine as you showed above

### Case 2: Export to a totally new directory `EXPORT_DIR`
- change the `dataset2` export above to save to `EXPORT_DIR`:
```
EXPORT_DIR = "/tmp/coco_export"
dataset2.export(
    export_dir=EXPORT_DIR,
    dataset_type=fo.types.ImageSegmentationDirectory,
    labels_path="segmentations",
    export_media=False,
)
```

#### Result
```
# output path empty
$ ls /tmp/coco_export/segmentations/

# the data folder of the TEST_DIR is unwanted populated with the segmentation masks
$ls /tmp/segmentations/data
000000000139.jpg  000000000285.jpg  000000000632.jpg  000000000724.jpg
000000000139.png  000000000285.png  000000000632.png  000000000724.png
...
```

Hope you reproduce it now.

**Comment by brimoor** [comment]
Hmm, I'm still not following what output you're looking for. Here's all the variations I can think of:

1. Export images only:

```py
dataset2.export(
    export_dir="/tmp/images",
    dataset_type=fo.types.ImageDirectory,
)
```

![000000005193](https://user-images.githubusercontent.com/25985824/190414823-87805b8f-656b-48ad-b141-801492add7ea.jpg)

2. Export masks only:

```py
dataset2.export(
    labels_path="/tmp/masks",
    dataset_type=fo.types.ImageSegmentationDirectory,
    export_media=False,
)
```

![000000005193](https://user-images.githubusercontent.com/25985824/190414858-96dec695-582b-42e4-a5d2-aada8c058208.png)

3. Render masks as colored overlays on the images themselves ([docs](https://voxel51.com/docs/fiftyone/user_guide/draw_labels.html)):

```py
dataset2.draw_labels("/tmp/rendered-masks")
```

![000000005193](https://user-images.githubusercontent.com/25985824/190414900-5c3c7590-78c8-43e9-9d01-1485712167e7.jpg)

**Comment by dahliarobotics** [comment]
I want to do export masks only to a location different than the original dataset (what you showed as 2. point above). However, as pointed out in my last comment (under Case 2), when doing this (using `export_media=False`) my original dataset directory gets messed up and the goal directory stays empty.
But seems like you can not reproduce this problem...

**Comment by brimoor** [comment]
Ah thank you, now I understand! I am able to reproduce this bug on `fiftyone==0.16.5` and have verified that it will be fixed in `fiftyone==0.17.0` which will be released in the next couple days.

In the meantime, you can achieve the output you'd like on your existing install by passing an absolute path to `labels_path` rather than providing both `export_dir` and a relative `labels_path`:

```py
dataset2.export(
    dataset_type=fo.types.ImageSegmentationDirectory,
    labels_path="/tmp/coco_export/segmentations",
    export_media=False,
)
```

It seems that the current FiftyOne release is getting confused when trying to join `export_dir` and `labels_path` when the syntax you're trying is in use.
